### PR TITLE
feat: improvements to add container cmd

### PR DESCRIPTION
**Pull request by kirederik**
this pr includes three commits:

- **feat: add container to include Dockerfile**
    - the command currently adds the Dockerfile and a script, but the Dockerfile
      was empty.
    - also: instead of generating file with contents by hand, i moved the
      contents to a template file.
- **fix: move templating logic to root.go**
    - small chore to concentrate the templating logic in one place; there's
      probably other places that i can update to use the new function, but kept
      it simple for now

### CODE REVIEW

File: cmd/add_container.go [hunk 1]
```diff
@@ -154,14 +158,12 @@ func AddContainer(cmd *cobra.Command, args []string) error {
 			return err
 		}
 	}
-
-	err = generatePipelineDirFiles(workflowDirectory, pipelineName, containerName)
-	if err != nil {
+	if err := generatePipelineDirFiles(dir, workflowPath, pipelineName, containerName); err != nil {
+		fmt.Println(dir, workflowPath, pipelineName, containerName)
```
Comment (abangser) [inline cmd/add_container.go:162]: What does this println do for us? was this a debug line or are we hoping for a error message here?

**Review by abangser — COMMENTED** [review]
Confirmed image builds and runs as expected 🎉 

There does seem to be a switch in the naming convention of the directories which surprised me. On an older version my directory structure was:
```
local-old
├── README.md
├── example-resource.yaml
├── promise.yaml
└── workflows
    └── resource
        └── configure
            └── instance
                ├── Dockerfile
                ├── resources
                └── scripts
                    └── pipeline.sh
```

and on this PR I see:
```

local-new
├── README.md
├── example-resource.yaml
├── promise.yaml
└── workflows
    └── resource
        └── configure
            └── instance
                **└── blah**
                    ├── Dockerfile
                    ├── resources
                    └── scripts
                        └── pipeline.sh
```

I don't see where that got introduced in this change though, so not sure if it came in between this PR and the version I had locally from Wednesday.

File: cmd/add_container.go (hunk 1 above)
Comment (kirederik) [reply cmd/add_container.go:162]: def a debug line, good catch!

**Comment by kirederik** [comment]
> Confirmed image builds and runs as expected 🎉
> 
> There does seem to be a switch in the naming convention of the directories which surprised me. On an older version my directory structure was:
> 
> ```
> local-old
> ├── README.md
> ├── example-resource.yaml
> ├── promise.yaml
> └── workflows
>     └── resource
>         └── configure
>             └── instance
>                 ├── Dockerfile
>                 ├── resources
>                 └── scripts
>                     └── pipeline.sh
> ```
> 
> and on this PR I see:
> 
> ```
> 
> local-new
> ├── README.md
> ├── example-resource.yaml
> ├── promise.yaml
> └── workflows
>     └── resource
>         └── configure
>             └── instance
>                 **└── blah**
>                     ├── Dockerfile
>                     ├── resources
>                     └── scripts
>                         └── pipeline.sh
> ```
> 
> I don't see where that got introduced in this change though, so not sure if it came in between this PR and the version I had locally from Wednesday.

`blah` is the name of the container for that pipeline; it was introduced by https://github.com/syntasso/kratix-cli/pull/16 (hence you can't see why change in this commit 😆)

**Pull request merged**